cmd/demo-client/main.go

```diff
@@ -159,3 +206,53 @@ func Watch(c *cli.Context, client client.Client) error {
 	}
 	return nil
 }
+
+func newPrometheusBridge(address string, gateway string, pushIntervalSec int64) prometheus.Registerer {
+	b := &prometheusBridge{
+		address:         address,
+		pushIntervalSec: pushIntervalSec,
+		registry:        prometheus.NewRegistry(),
+	}
+	if gateway != "" {
+		b.pusher = push.New(gateway, "drand_client_observations_push")
+		go b.pushLoop()
+	}
+	if address != "" {
+		http.Handle("/metrics", promhttp.HandlerFor(b.registry, promhttp.HandlerOpts{
+			Timeout: 10 * time.Second,
+		}))
+		go log.Fatal(http.ListenAndServe(address, nil))
+	}
+	return b
+}
+
+type prometheusBridge struct {
+	address         string
+	pushIntervalSec int64
+	registry        *prometheus.Registry
+	pusher          *push.Pusher
+}
+
+func (b *prometheusBridge) pushLoop() {
+	for {
+		time.Sleep(time.Second * time.Duration(b.pushIntervalSec))
+		if err := b.pusher.Push(); err != nil {
+			log.Printf("prometheus gateway push (%v)", err)
+		}
+	}
+}
+
+func (b *prometheusBridge) Register(x prometheus.Collector) error {
```

Comment:
if the first member of the struct is the `Registry`, then the registry can directly implement these methods.
In the constructor, the pusher can use [`Gatherer`](https://godoc.org/github.com/prometheus/client_golang/prometheus/push#Pusher.Gatherer) to push whatever has been registered with the registry.

Reply:
Done.